airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/jsonl/JsonLSerializedBuffer.java

```diff
@@ -53,14 +54,19 @@ protected void writeRecord(final AirbyteRecordMessage record) {
     json.put(JavaBaseConstants.COLUMN_NAME_AB_ID, UUID.randomUUID().toString());
     json.put(JavaBaseConstants.COLUMN_NAME_EMITTED_AT, record.getEmittedAt());
     if (flattenData) {
-      Map<String, JsonNode> data = MAPPER.convertValue(record.getData(), new TypeReference<>() {});
+      final Map<String, JsonNode> data = MAPPER.convertValue(record.getData(), new TypeReference<>() {});
       json.setAll(data);
     } else {
       json.set(JavaBaseConstants.COLUMN_NAME_DATA, record.getData());
     }
     printWriter.println(Jsons.serialize(json));
   }
 
+  @Override
+  protected void writeRecord(final String recordString, final long emittedAt) throws IOException {
```

Comment:
I see what you mean now, it has the same pattern as CSVwriter. I'm going to opt to add a TODO here since that cuts into the critical path and want to keep these changes isolated to AsyncConsumer